pkg/remote/builtin/github/github.go

```diff
@@ -34,12 +37,14 @@ type GitHub struct {
 
 func New(conf *config.Config) *GitHub {
```

Comment:
yep, exactly!

so a remote package (ie `/pkg/remote/github`) would register a `DriverFunc`:

```Go
func init() {
	remote.Register(NewGithubDriver)
}

func NewGithubDriver() (remote.Remote, error) {
	// return github.GitHub with data populated from environment variables
}
```

The `remote` package would keep track of all registered drivers:

```Go
type DriverFunc func() (Remote, error)

func Register(name string, driver DriverFunc) {
	if driver == nil {
		panic("remote: Register driver is nil")
	}
	if _, dup := drivers[name]; dup {
		panic("remote: Register called twice for remote driver " + name)
	}
	drivers[name] = driver
}
```

So when Drone starts we could do something like:

```Go
remote.Load(os.GetEnv("REMOTE_DRIVER"))
```